engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -614,7 +615,16 @@ protected void serviceStart() throws Exception
       }
       WebApp webApp = WebApps.$for("stram", StramAppContext.class, appContext, "ws").with(config).start(new StramWebApp(this.dnmgr));
       LOG.info("Started web service at port: " + webApp.port());
-      this.appMasterTrackingUrl = NetUtils.getConnectAddress(webApp.getListenerAddress()).getHostName() + ":" + webApp.port();
+      String host = NetUtils.getConnectAddress(webApp.getListenerAddress()).getHostName() + ":" + webApp.port();
+
+      // For backward compatibility, not adding scheme in TrackingURL for non-HTTPS
+      // TODO: Remove the check in next major release and add scheme always
+      if (ConfigUtils.isSSLEnabled(config)) {
+        String scheme = ConfigUtils.getSchemePrefix(config);
```

Comment:
@pradeepdalvi : @vrozov had answered your question. Calling getSchemePrefix is again calling isSSLEnabled. Why is that requried